### ISSUE TITLE
debian 0.6.4-1 to 0.6.4-1b

### ISSUE THREAD

**Issue by ggzengel** [zfsonlinux/pkg-zfs#154]
Is there a docu about changes?

**Comment by ggzengel** [comment]
What are the changes between 0.6.4-1.1-1 and 0.6.4-1.1-2?

**Comment by FransUrbo** [comment]
This is a not a support forum, please take this to the list.

Any changes in and between versions is described in the changes file that accompanies the packages.

```
Changes: 
 zfs-linux (0.6.4-1.1-2-wheezy) wheezy; urgency=low
 .
   * Make sure we only get 'filesystem's when we're about to mount it/them.
     This to filter out volumes (i.e. swap etc) and snapshots.
     Thanx to 'kidmin' for fix.
```

**Comment by ggzengel** [comment]
aptitude show me:
You can only view changelogs of official Debian packages

**Comment by ggzengel** [comment]
My question still exists:
How to get changelog before installing packages?

**Comment by ggzengel** [comment]
2nd question:
When comes 0.6.4-2?

**Comment by FransUrbo** [comment]
> How to get changelog before installing packages?
> 
> I honestly don't know then. Because the mail admin keeps blocking me (and others) and keeps 'throwing me out' from the mailing lists (especially the announce list where I used to announce these things), I don't know where to get this now. Not without going through some hoops anyway:

```
apt-get source zfs-linux=0.6.4-1.2-1-wheezy
```

Then look in the "zfs-linux-*/debian/changelog" file. Just replace the version with the version you're interested in…

The other way is just keeping a tab on the commits in the pkg-{zfs,spl} repos.

AFTER installation though, the change log is at

```
 /usr/share/doc/libzfs2/changelog.Debian.gz
```

so it depends a little on WHY you want to know what changed. If the "only" thing you want/need to know is if you must reboot, then that is good enough.

A point release "inside" of a released version (such as 0.6.4.1, 0.6.4.2 etc, where 0.6.4 is the released version) will NEVER introduce any API change etc, so it's safe to run with the (slightly) older module with the newer userland.

> 2nd question:
> When comes 0.6.4-2?
> 
> I released the package on "Sat Jun 27 15:06:23 2015" as version "0.6.4-1.2-1". I know the versions doesn't make much sense, but it would clash with the dailies otherwise. For 0.6.5, I have to come up with something else…

**Comment by FransUrbo** [comment]
> How to get changelog before installing packages?
> 
> Actually, you can also use the command "apt-listchanges":

```
apt-listchanges -a /var/cache/apt/archives/zfsutils_0.6.4-21-53b1d9-wheezy_amd64.deb
```

This would give the the FULL changelog on the latest dailies package. You'd need to run something like:

```
apt-get clean
apt-get update
apt-get -d dist-upgrade
```

first to get all the packages apt wants to upgrade downloaded first though.

Then, if/when you're satisfied, just remove the "-d" option from the dist-upgrade line.
